### PR TITLE
GRPC OnlineQuery (single)

### PR DESCRIPTION
**Pull request by rooftoofwoof**
Add non-bulk GRPC online query. Implemented via GRPC bulk query underneath. 

**Limitations**
- [x] Each `FeatureResult` has no `FeatureResolutionMeta`, `Pkey`, `QueryTimestamp`, `Error` (errors exist at top level)

### CODE REVIEW

File: client_grpc.go [hunk 1]
```diff
@@ -308,6 +305,57 @@ func (c *clientGrpc) OnlineQueryBulk(args OnlineQueryParamsComplete) (OnlineQuer
...
+	args.underlying.inputs = bulkInputs
+
+	bulkRes, err := c.onlineQueryBulk(args)
+	if err != nil {
+		// intentionally don't wrap, original error is good enough
+		return OnlineQueryResult{}, err
+	}
+
+	if resultHolder != nil {
+		if err := bulkRes.UnmarshalInto([]any{resultHolder}); err != nil {
```
Comment (rooftoofwoof) [inline client_grpc.go:324]: add test for this, particularly

File: client_grpc.go [hunk 2]
```diff
@@ -308,6 +307,122 @@ func (c *clientGrpc) OnlineQueryBulk(args OnlineQueryParamsComplete) (OnlineQuer
...
+		slice := reflect.MakeSlice(reflect.SliceOf(reflect.TypeOf(singleValue)), 1, 1)
+		slice.Index(0).Set(reflect.ValueOf(singleValue))
+		bulkInputs[k] = slice.Interface()
+	}
+	args.underlying.inputs = bulkInputs
+
+	bulkRes, err := c.onlineQueryBulk(args)
+	if err != nil {
+		// intentionally don't wrap, original error is good enough
+		return OnlineQueryResult{}, err
```
Comment (rooftoofwoof) [inline client_grpc.go:322]: return bulkRes, err

File: client_grpc.go [hunk 3]
```diff
@@ -308,6 +307,122 @@ func (c *clientGrpc) OnlineQueryBulk(args OnlineQueryParamsComplete) (OnlineQuer
...
+			continue
+		}
+
+		colNames := lo.Keys(rowsHm[0])
+		colValues := make([][]any, 0, len(rowsHm))
+		for _, col := range colNames {
+			vals := lo.Map(rowsHm, func(row map[string]any, _ int) any {
+				return row[col]
+			})
+			colValues = append(colValues, vals)
```
Comment (rooftoofwoof) [inline client_grpc.go:391]: inline

File: internal/tests/integration/query_test.go [hunk 4]
```diff
@@ -11,46 +12,60 @@ import (
...
+
+			var explicitUser user
+			err = res.UnmarshalInto(&explicitUser)
+			// TODO: We need to fix this nil check
+			//       to just be `!= nil`
+			if err != (*chalk.ClientError)(nil) {
+				t.Fatal("Failed unmarshaling result", err)
+			}
 
+			for _, testUser := range []user{implicitUser, explicitUser} {
```
Comment (rooftoofwoof) [inline internal/tests/integration/query_test.go:59]: make clear whether we're testing implicit or explicit?

File: client_grpc.go [hunk 5]
```diff
@@ -308,6 +320,137 @@ func (c *clientGrpc) OnlineQueryBulk(args OnlineQueryParamsComplete) (OnlineQuer
...
+	if resultHolder != nil {
+		// Create a pointer to an empty slice and unmarshal into that
+		var holderType = reflect.TypeOf(resultHolder)
+		if holderType.Kind() != reflect.Ptr {
+			return OnlineQueryResult{}, errors.Newf(
+				"result holder must be a pointer, found %s",
+				holderType.Kind(),
+			)
+		}
+		var structType = holderType.Elem()
```
Comment (rooftoofwoof) [inline client_grpc.go:347]: what's this - just use `:=` lol

**Review by AndyMoreland — APPROVED** [review]

**Pull request merged**